packages/sdk/webpack.config.ts

```diff
@@ -6,48 +6,81 @@ import url from 'url';
 
 const __dirname = url.fileURLToPath(new URL('.', import.meta.url));
 
-const config: webpack.Configuration = {
-    mode: 'production',
-    cache: {
-        type: 'filesystem',
-        cacheDirectory: resolve(__dirname, '.webpack-cache'),
-    },
-    entry: {
-        concordium: resolve(__dirname, 'src/index.ts'),
-    },
-    plugins: [
-        new webpack.SourceMapDevToolPlugin({
-            filename: '[file].map',
-        }),
-    ],
-    resolve: {
-        extensionAlias: {
-            '.js': ['.ts', '.js'],
+function configFor(
+    target: 'web' | 'node' | 'react-native'
+): webpack.Configuration {
+    const t = target === 'react-native' ? 'web' : target;
+    const entry =
+        target === 'react-native'
+            ? [
+                  resolve(__dirname, './polyfill/react-native.ts'),
```

Comment:
**Suggestion:** Separate the polyfills into a subpath.

I believe it is bad practice for a library to silently enable polyfills in your environment and we should at least split this into a separate file and instruct the user can enable it for react native.

Reply:
Resolving subpath exports is not stable in metro. I guess we could instruct users to do either

```ts
import "@concordium/web-sdk/react-native-compat"; // subpath export
import "@concordium/web-sdk/lib/esm/pub/react-native-compat"; // deep import
```

Support for subpath export is expected to be stable in [release 0.73 in Q3 2023](https://reactnative.dev/blog/2023/06/21/package-exports-support#the-future-stable-exports-enabled-by-default), so I guess it would be fine to work towards this.

Reply:
Will push polyfill responsibility to consumer